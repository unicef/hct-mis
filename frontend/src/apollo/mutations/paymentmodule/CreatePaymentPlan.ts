import { gql } from 'apollo-boost';

export const CREATE_PAYMENT_PLAN = gql`
  mutation CreatePP($input: CreatePaymentPlanInput!) {
    createPaymentPlan(input: $input) {
      paymentPlan {
        id
      }
    }
  }
`;
