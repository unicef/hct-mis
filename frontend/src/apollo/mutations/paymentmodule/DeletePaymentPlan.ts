import { gql } from 'apollo-boost';

export const DELETE_PAYMENT_PLAN = gql`
  mutation DeletePP($paymentPlanId: ID!) {
    deletePaymentPlan(paymentPlanId: $paymentPlanId) {
      paymentPlan {
        id
        status
      }
    }
  }
`;
