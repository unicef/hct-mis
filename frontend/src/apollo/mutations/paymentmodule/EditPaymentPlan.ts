import { gql } from 'apollo-boost';

export const UPDATE_PAYMENT_PLAN = gql`
  mutation UpdatePP($input: UpdatePaymentPlanInput!) {
    updatePaymentPlan(input: $input) {
      paymentPlan {
        id
      }
    }
  }
`;
