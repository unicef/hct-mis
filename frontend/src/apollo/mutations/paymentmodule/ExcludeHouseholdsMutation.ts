import { gql } from 'apollo-boost';

export const EXCLUDE_HOUSEHOLDS_PP = gql`
  mutation ExcludeHouseholdsPP($input: //TODO!!!) {
    excludeHouseholds(input: $input) {
      paymentPlan {
        id
        excludedHouseholdsIds
    }
  }
`;
