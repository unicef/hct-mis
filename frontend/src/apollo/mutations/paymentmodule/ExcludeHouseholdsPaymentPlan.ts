import { gql } from 'apollo-boost';

export const EXCLUDE_HOUSEHOLDS_PP = gql`
  mutation ExcludeHouseholdsPP(
    $paymentPlanId: ID!
    $excludedHouseholdsIds: [String]!
  ) {
    excludeHouseholds(
      paymentPlanId: $paymentPlanId
      excludedHouseholdsIds: $excludedHouseholdsIds
    ) {
      paymentPlan {
        id
        excludedHouseholds {
          id
          unicefId
        }
      }
    }
  }
`;
