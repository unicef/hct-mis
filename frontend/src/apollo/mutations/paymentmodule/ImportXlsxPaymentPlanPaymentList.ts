import { gql } from 'apollo-boost';

export const ImportXlsxPPList = gql`
  mutation importXlsxPPList($paymentPlanId: ID!, $file: Upload!) {
    importXlsxPaymentPlanPaymentList(
      paymentPlanId: $paymentPlanId
      file: $file
    ) {
      paymentPlan {
        id
        status
      }
      errors {
        sheet
        coordinates
        message
      }
    }
  }
`;
