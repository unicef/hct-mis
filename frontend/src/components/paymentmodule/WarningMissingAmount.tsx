import { Box } from '@material-ui/core';
import WarningIcon from '@material-ui/icons/Warning';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): React.ReactElement {
  const { t } = useTranslation();

  return (
    <Box display='flex' alignItems='center'>
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
