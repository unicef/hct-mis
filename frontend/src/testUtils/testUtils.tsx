import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
