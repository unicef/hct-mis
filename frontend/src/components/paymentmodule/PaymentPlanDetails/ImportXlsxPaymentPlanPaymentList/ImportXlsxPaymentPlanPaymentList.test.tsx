import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { fakeImportXlsxPpListMutation } from '../../../../../fixtures/paymentmodule/fakeImportXlsxPpListMutation';
import { render } from '../../../../testUtils/testUtils';
import { ImportXlsxPaymentPlanPaymentList } from './ImportXlsxPaymentPlanPaymentList';

describe('components/paymentmodule/PaymentPlanDetails/ImportXlsxPaymentPlanPaymentList', () => {
  it('should render', async () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={fakeImportXlsxPpListMutation}>
        <ImportXlsxPaymentPlanPaymentList paymentPlan={fakeApolloPaymentPlan} />
      </MockedProvider>,
    );

    const buttonImport = container.querySelector('[data-cy="button-import"]');
    expect(buttonImport).toBeInTheDocument();

    act(() => {
      buttonImport.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    const inputFile = document.querySelector<HTMLInputElement>('input[type="file"]');
    const buttonImportSubmit = document.querySelector('[data-cy="button-import-submit"]');
    const buttonClose = document.querySelector('[data-cy="close-button"]');

    expect(inputFile).toBeInTheDocument();
    expect(buttonImportSubmit).toBeInTheDocument();
    expect(buttonClose).toBeInTheDocument();

    const file = new File([new ArrayBuffer(210715200)], 'sample.xlsx', {
      type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    });
    Object.defineProperty(inputFile, 'files', {
      value: [file],
    });

    act(() => {
      inputFile.dispatchEvent(new Event('change', { bubbles: true }));
    });

    act(() => {
      buttonImportSubmit.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    act(() => {
      buttonClose.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    await act(() => wait(0)); // wait for response

    expect(container).toMatchSnapshot();
  });
});
