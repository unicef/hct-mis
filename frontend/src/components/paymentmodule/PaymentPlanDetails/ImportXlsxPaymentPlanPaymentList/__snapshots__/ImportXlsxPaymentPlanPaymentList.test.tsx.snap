// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/ImportXlsxPaymentPlanPaymentList should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary Mui-disabled Mui-disabled"
      data-cy="button-import"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-bZQynM bKNwua"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
        </span>
        Import XLSX
      </span>
    </button>
  </div>
</div>
`;
